packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -136,32 +138,54 @@ export class CacheDataClient implements IDataClient {
       `Creating cache client using endpoint: '${this.credentialProvider.getCacheEndpoint()}'`
     );
 
+    const channelConfig: ChannelConfiguration =
+      this.configuration.getChannelConfiguration();
+
+    const channelOptions: Record<string, number> = {
+      'grpc-node.max_session_memory': channelConfig.maxSessionMemoryMB,
+      'grpc.use_local_subchannel_pool': channelConfig.useLocalSubchannelPool
+        ? 1
+        : 0,
+    };
+
+    // Apply keepalive settings only if useKeepAlive is true
+    if (channelConfig.useKeepAlive) {
+      // Assert that keepalive configurations are not undefined
+      if (
+        channelConfig.keepAlivePermitWithoutCalls === undefined ||
+        channelConfig.keepAliveTimeoutMs === undefined ||
+        channelConfig.keepAliveTimeMs === undefined
+      ) {
+        throw new Error(
+          'Keepalive settings must be defined when useKeepAlive is enabled.'
+        );
```

Comment:
tbh i'd rather make these grpc settings basically just pass-through; i don't think you need an additional `useKeepAlive`. If you really want to do that, I would make it an interface that contains the nested settings.